fake.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// FakeCallArgs holds all the args passed in a call.
+type FakeCallArgs map[string]interface{}
+
+// FakeCall records the name of a called function and the passed args.
+type FakeCall struct {
+	// Receiver is the fake for which the function was called. It is
+	// not required, particularly if the function is not a method.
+	Receiver interface{}
+
+	// Funcname is the name of the function that was called.
+	FuncName string
+
+	// Args is the set of arguments to the function.
+	Args FakeCallArgs
+}
+
+// Fake is used in testing to stand in for some other value, to record
+// all calls to faked methods/functions, and to allow users to set the
+// values that are returned from those calls. Fake is intended to be
+// embedded in another struct that will define the methods to track:
+//
+//    type fakeConn struct {
+//        *testing.Fake
+//        Response []byte
+//    }
+//
+//    func newFakeConn() *fakeConn {
+//        return &fakeConn{
+//            Fake: &testing.Fake{},
+//        }
+//    }
+//
+//    func (fc fakeConn) Send(request string) []byte {
+//        fc.AddCall("Send", FakeCallArgs{
```

Comment:
Ugh! I wish this was more like this:

fc.Call("MyFunc", request)
and
fc.MethodCall(receiver, "MyMethod", arg1, arg2)

Why the complexity of having to add long map literals for recording calls?